testsuite/pytests/test_eprop_plasticity.py

```diff
@@ -0,0 +1,731 @@
+    spike_probability = duration["step"] * input_spike_rate
+    dtype_in_spks = np.float32
+
+    input_spike_bools = np.random.rand(n_batch, steps["sequence"], n_in) < spike_probability
+    input_spike_bools = np.hstack(input_spike_bools.swapaxes(1, 2))
+    input_spike_bools[:, 0] = 0
+
+    sequence_starts = np.arange(0.0, duration["task"], duration["sequence"])
+    params_gen_spk_in = []
+    for input_spike_bool in input_spike_bools:
+        input_spike_times = np.arange(duration["sequence"] * n_batch)[input_spike_bool]
+        input_spike_times_all = [input_spike_times + start for start in sequence_starts]
+        params_gen_spk_in.append({"spike_times": np.hstack(input_spike_times_all).astype(dtype_in_spks).tolist()})
+
+    nest.SetStatus(gen_spk_in, params_gen_spk_in)
+
+    # Create output
+
+    def generate_superimposed_sines(steps_sequence, periods):
+        n_sines = len(periods)
+
+        amplitudes = np.random.uniform(low=0.5, high=2.0, size=n_sines)
+        phases = np.random.uniform(low=0.0, high=2.0 * np.pi, size=n_sines)
+
+        sines = [
+            A * np.sin(np.linspace(phi, phi + 2.0 * np.pi * (steps_sequence // T), steps_sequence))
+            for A, phi, T in zip(amplitudes, phases, periods)
+        ]
+
+        superposition = sum(sines)
+        superposition -= superposition[0]
+        superposition /= max(np.abs(superposition).max(), 1e-6)
+        return superposition
+
+    target_signal = generate_superimposed_sines(steps["sequence"], [1000.0, 500.0, 333.0, 200.0])
+
+    params_gen_rate_target = {
+        "amplitude_times": (np.arange(0, duration["task"], duration["step"]) + duration["step"]).tolist(),
+        "amplitude_values": np.tile(target_signal, n_iter * n_batch).tolist(),
+    }
+
+    nest.SetStatus(gen_rate_target, params_gen_rate_target)
+
+    # Simulate
+
+    nest.Simulate(duration["sim"])
+
+    # Read out recorders
+
+    events_mm_out = mm_out.get("events")
+
+    # Evaluate training error
+
+    readout_signal = events_mm_out["readout_signal"]
+    target_signal = events_mm_out["target_signal"]
+
+    readout_signal = readout_signal[steps["shift"] - 1 :]
+    target_signal = target_signal[steps["shift"] - 2 : -1]
+
+    error = (readout_signal - target_signal) ** 2
+    loss = 0.5 * np.add.reduceat(error, np.arange(0, steps["task"], steps["sequence"]))
+
+    # Verify results
+
+    loss_NEST_verification = np.array(
+        [
+            0.301624405492225800,
+            0.300969013603013313,
+            0.295005356673212760,
+            0.299270730552683328,
+            0.320109501410700492,
+        ]
+    )
+
+    loss_TF_verification = np.array(
+        [
+            0.301624119281768799,
+            0.300968766212463379,
+            0.295005142688751221,
+            0.299270510673522949,
+            0.320109188556671143,
+        ]
+    )
+
+    assert np.all(loss == loss_NEST_verification)
+    assert np.all(loss - loss_TF_verification < 1e-6)
```

Comment:
np.allclose() would be slightly more elegant here.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/2867/commits/d7b5ce344ab87102fbe75f41820f2937ca68af7d